src/sentry/workflow_engine/handlers/detector/base.py

```diff
@@ -34,6 +34,7 @@ class EvidenceData(Generic[DataPacketEvaluationType]):
     detector_id: int
     data_packet_source_id: int
     conditions: list[dict[str, Any]]
+    data_source_definition: dict[str, Any] | None
```

Comment:
Also wanted to call out that I'm adding a property to this dataclass and I know python will raise an exception when instantiating if it doesn't exist. Is it safe to do this? Want to make sure this won't start failing for old event data when this is merged

Reply:
🤔 it should be okay, but might want to set a default value to `None` to be safe.

afaik, we only apply this dataclass to the write on an issue occurrence, and when we're fetching for the API we just pass it through the serializer.

Reply:
Okay added a default value, but I agree I _think_ it would probably be okay without one